### PR TITLE
POM-208 - dodanie paginacji do zapytan.

### PR DESCRIPTION
**Pull request by oleksandrdevqube**

### CODE REVIEW

File: src/app/shared/components/paginator/paginator.component.html [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<div class="container">
+  <mat-paginator
+    class="paginator"
+    [length]="length"
+    [pageSizeOptions]="[5, 10, 20, 59]"
+    aria-label="Select page"
```
Comment (neu5) [inline src/app/shared/components/paginator/paginator.component.html:6]: translation key?

File: src/app/shared/components/paginator/paginator.component.scss [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      .mat-paginator-page-size-label {
+        font-size: 15px;
+        color: var(--c-darkblue);
+      }
+
+      .mat-form-field-flex {
+        padding: 0.3em 0.75em 0.4em 0.75em;
+
+        .mat-form-field-outline {
+          background: white;
```
Comment (neu5) [inline src/app/shared/components/paginator/paginator.component.scss:17]: var(--c-white);

File: src/app/shared/components/paginator/paginator.component.html [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<div class="container">
+  <mat-paginator
+    class="paginator"
+    [length]="length"
+    [pageSizeOptions]="[5, 10, 20, 59]"
```
Comment (neu5) [inline src/app/shared/components/paginator/paginator.component.html:5]: for sure 59? not 50?

**Review by neu5 — APPROVED** [review]

File: src/app/find-help/accommodation-search/accommodation-search.component.ts [hunk 4]
```diff
@@ -29,17 +31,20 @@ export class AccommodationSearchComponent {
...
   }
 
-  search(searchCriteria: AccommodationQuery) {
+  search(searchCriteria?: AccommodationQuery, pagination?: Pageable) {
     this.loading = true;
 
-    // TODO specify the params; at least sorting is configurable from the UI
+    this.searchCriteria.capacity = searchCriteria?.capacity || this.searchCriteria.capacity;
+    this.searchCriteria.location = searchCriteria?.location || this.searchCriteria.location;
+    this.pagination = pagination || this.pagination;
```
Comment (mgol) [inline src/app/find-help/accommodation-search/accommodation-search.component.ts:39]: `||` is dangerous as `0` is often a valid value. Always use `??` for such fallback logic which only catches `null` and `undefined`.

```suggestion
    this.searchCriteria.capacity = searchCriteria?.capacity ?? this.searchCriteria.capacity;
    this.searchCriteria.location = searchCriteria?.location ?? this.searchCriteria.location;
    this.pagination = pagination ?? this.pagination;
```

File: src/app/find-help/material-aid-search/material-aid-search.component.ts [hunk 5]
```diff
@@ -13,18 +13,24 @@ export class MaterialAidSearchComponent {
...
+  pagination: Pageable | undefined = {};
   constructor(private materialAidResourceService: MaterialAidResourceService) {}
 
-  search(searchCriteria: MaterialAidOfferSearchCriteria) {
+  search(searchCriteria?: MaterialAidOfferSearchCriteria, pagination?: Pageable) {
     this.loading = true;
 
+    this.searchCriteria.category = searchCriteria?.category || this.searchCriteria.category;
+    this.searchCriteria.location = searchCriteria?.location || this.searchCriteria.location;
+    this.pagination = pagination || this.pagination;
```
Comment (mgol) [inline src/app/find-help/material-aid-search/material-aid-search.component.ts:25]:
```suggestion
    this.searchCriteria.category = searchCriteria?.category ?? this.searchCriteria.category;
    this.searchCriteria.location = searchCriteria?.location ?? this.searchCriteria.location;
    this.pagination = pagination ?? this.pagination;
```

File: src/app/find-help/transport-search/transport-search.component.ts [hunk 6]
```diff
@@ -13,19 +13,26 @@ export class TransportSearchComponent {
...
-  search(searchCriteria: TransportOfferSearchCriteria) {
+  search(searchCriteria?: TransportOfferSearchCriteria, pagination?: Pageable) {
     this.loading = true;
 
-    // TODO specify the params; at least sorting is configurable from the UI
+    this.searchCriteria.origin = searchCriteria?.origin || this.searchCriteria.origin;
+    this.searchCriteria.destination = searchCriteria?.destination || this.searchCriteria.destination;
+    this.searchCriteria.capacity = searchCriteria?.capacity || this.searchCriteria.capacity;
+    this.searchCriteria.transportDate = searchCriteria?.transportDate || this.searchCriteria.transportDate;
+    this.pagination = pagination || this.pagination;
```
Comment (mgol) [inline src/app/find-help/transport-search/transport-search.component.ts:27]:
```suggestion
    this.searchCriteria.origin = searchCriteria?.origin ?? this.searchCriteria.origin;
    this.searchCriteria.destination = searchCriteria?.destination ?? this.searchCriteria.destination;
    this.searchCriteria.capacity = searchCriteria?.capacity ?? this.searchCriteria.capacity;
    this.searchCriteria.transportDate = searchCriteria?.transportDate ?? this.searchCriteria.transportDate;
    this.pagination = pagination ?? this.pagination;
```

**Review by mgol — APPROVED** [review]

**Pull request merged**